server/service/service_appconfig.go

```diff
@@ -128,8 +129,10 @@ func cleanupURL(url string) string {
 }
 
 func (svc *Service) License(ctx context.Context) (*fleet.LicenseInfo, error) {
-	if err := svc.authz.Authorize(ctx, &fleet.AppConfig{}, fleet.ActionRead); err != nil {
-		return nil, err
+	if !svc.authz.IsAuthenticatedWith(ctx, authz_ctx.AuthnDeviceToken) {
+		if err := svc.authz.Authorize(ctx, &fleet.AppConfig{}, fleet.ActionRead); err != nil {
+			return nil, err
+		}
 	}
```

Comment:
Given how rego currently handles this (any user is allowed to read global config), this is probably ok.

In the future we should define a "license" object in rego (as it has nothing to do with app config)

Reply:
> In the future we should define a "license" object in rego (as it has nothing to do with app config)
@lucasmrod, totally agree. It really threw me off when I first saw that the auth was based on app config rather than a distinct rego object type for license info. Shall I open a ticket on this? one